pkg/util/k8sutil/tpr_util.go

```diff
@@ -74,16 +78,33 @@ func GetClusterTPRObject(restcli rest.Interface, ns, name string) (*spec.Cluster
 	return readOutCluster(b)
 }
 
-// UpdateClusterTPRObject updates the given TPR object.
-// ResourceVersion of the object MUST be set or update will fail.
-func UpdateClusterTPRObject(restcli rest.Interface, ns string, c *spec.Cluster) (*spec.Cluster, error) {
-	if len(c.Metadata.ResourceVersion) == 0 {
-		return nil, errors.New("k8sutil: resource version is not provided")
-	}
-	return updateClusterTPRObject(restcli, ns, c)
+// AtomicUpdateClusterTPRObject takes an update function which is executed before attempting
+// to update the Cluster TPR object. The entire process is retried
+// again until the update is successful or a non-conflict error is returned.
+func AtomicUpdateClusterTPRObject(restcli rest.Interface, name, namespace string, updateFunc ClusterTPRModifier) (*spec.Cluster, error) {
+	var updatedCluster *spec.Cluster
+	err := retryutil.Retry(5*time.Second, 5, func() (done bool, err error) {
```

Comment:
No reason. 5 just seemed like a reasonable number though I can increase it to 10 or something. The point being we would only be retrying a limited number of times.